app/serializers/search/cms_serializer.rb

```diff
@@ -57,7 +57,7 @@ def sorted_pages
   end
 
   def date(page)
-    _date = cms_fragment_content(:published_date, page)
+    _date = cms_fragment_content_datetime(:published_date, page)
```

Comment:
Thanks for the comment Ferdi. There's a somewhat related point here about using `cms_fragment_content_datetime` in the CMS serializer - currently all the datetime fragments in my CMS layouts are `date_not_null` tags but there's the possibility that it might not be sometimes. In this case, would we want to continue returning `_date` even if it's `nil` (which is what the tertiary condition is checking for) ?

Reply:
@stanleypliu Good point. I think it might be useful to return check of the default date tag in case date_not_null is not used. Which is something you somewhat did previously and then removed from the content_datetime method :)

So perhaps your custom method could be something like:

```
 def cms_fragment_content_datetime(identifier, page)
    date = page.fragments.find_by(tag: 'date_not_null', identifier: identifier).try(:datetime)
    date ? date : page.fragments.find_by(tag: 'date', identifier: identifier).try(:datetime)
  end
```

So the above is very specific for dates kind of fragments. Otherwise, it could also be:

```
 def cms_fragment_content_datetime(identifier, page)
    date = page.fragments.find_by(tag: 'date_not_null', identifier: identifier).try(:datetime)
    date ? date : cms_fragment_content(identifier, page)
 end
```

which is more general, but if misused it could return the content of another tag, like a normal content string, which could make the strftime line below fail.

What do you think?

Reply:
Second method makes more sense to me - I originally intended this method to fall back to using the Comfy helper if it was passed an non-date tag.

Reply:
Yeah, if you are going for the second method I've written above that is a good idea!
Let me know when this is ready to be reviewed again :)